methods/run_multiplier_on_NF_samples.Rmd

```diff
@@ -0,0 +1,224 @@
+```{r}
+abundance <- tx_data$abundance %>% as.data.frame() 
+
+mapped_list <- mapIds(org.Hs.eg.db, keys = rownames(abundance), column = "SYMBOL", 
+                      keytype = "ENSEMBL", multiVals = "list")
+
+mapped_df <- reshape2::melt(mapped_list)
+
+colnames(mapped_df) <- c("SYMBOL", "ENSEMBL")
+
+abundance$gene_id <- rownames(abundance)
+
+# let's use the mappings to reannotate our data
+abundance <- mapped_df %>%
+  # removing anything in the SYMBOL column that is NA
+  filter(!is.na(SYMBOL)) %>%
+  # join by Ensembl gene ids, retaining only Ensembl gene ids in both
+  inner_join(y = abundance, by = c("ENSEMBL" = "gene_id")) %>% 
+  dplyr::select(-ENSEMBL) %>% 
+  add_count(SYMBOL)
+
+duplicate_genes <- abundance %>% filter(n > 1) %>% dplyr::select(-n)
+
+abundance_filt <- abundance  %>% 
+  filter(n == 1) %>% 
+  tibble::column_to_rownames("SYMBOL") %>% 
+  dplyr::select(-n)  %>%
+  as.matrix()
+
+abundance_filt_cutoff <- abundance_filt[rowSums(abundance_filt) > 10,]
+
+```
+
+# multiPLIER
+
+## Filtering out LVs
+
+```{r}
+mp_res <- GetNewDataB(abundance_filt_cutoff, plier_model)
+```
+
+### Filter by standard deviation 
+
+Most of the latent variables have low standard deviation across this cohort of samples, and thus are probably not interesting for comparing across or within tumor groups. A cutoff of SD > 0.075 seems to be a reasonable cutoff to filter out the least variable LVs. 
+
+```{r}
+mp_res_tidy <- mp_res %>% 
+  as.data.frame(.) %>% 
+  tibble::rownames_to_column('latent_var') %>% 
+  tidyr::gather("id","value",-latent_var) %>% 
+  group_by(latent_var) %>% 
+  mutate(sd_value = sd(value)) %>% 
+  ungroup() %>% 
+  mutate(trunc_label = stringr::str_trunc(latent_var, 10, "right")) %>% 
+  left_join(metadata)
+
+ggplot(data= mp_res_tidy %>% dplyr::select(latent_var,sd_value) %>% distinct()) +
+  geom_histogram(aes(x=sd_value))
+
+ggplot(data = mp_res_tidy) +
+  geom_boxplot(aes(x=reorder(trunc_label, -sd_value), y = value, color = sd_value > 0.075))
+
+```
+
+It is probably a safe assumption that (at least for subtyping these tumors), the latent variables with high variance across the cohort are the most "interesting." We won't filter these out a priori because we have some downstream analyses that might use te low sd data as well. 
+
+```{r eval=FALSE, include=FALSE}
+
+# mp_res_tidy_filt <- mp_res_tidy %>% 
+#   filter(sd_value > 0.05)
+# 
+# mp_res_filt <- mp_res[rownames(mp_res) %in% unique(mp_res_tidy_filt$latent_var),]
+
+```
+
+### Filter by Correlation
+
+We'll correlate the loadings of the latent variables to see if any have huge overlap between one another. In the plot below, if "high SD" is true, we are looking at the correlation of a latent variable where the standard deviation is 0.075 to another LV with a SD > 0.075. All other comparisions are in red. 
+
+```{r}
+
+plier_loadings <- plier_model$Z 
+colnames(plier_loadings) <- rownames(mp_res)
+
+plier_cor <- cor(plier_loadings) 
```

Comment:
is this pearson cor or spearman? it might be better to do a rank-based approach if you did pearson.

Reply:
This is pearson. Let's talk about this @ standup!

Reply:
(or sitdown)